mass_mailing_switzerland/models/account_invoice_line.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Nathan Fluckiger <nathan.fluckiger@hotmail.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import models, fields
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+    campaign_id = fields.Many2one('umt.campaign')
```

Comment:
Instead of adding the 3 relations you can simply inherit from `utm.mixin` object. This is exactly made for inheriting those 3 fields. You can do this like this:
```
_inherit = ['account.invoice.line', 'utm.mixin']
_name = 'account.invoice.line'
```
And you can remove the fields.

Reply:
I don't see anywhere where you set those values in the invoice.line ? I'm not sure those fields are actually needed?